assets/src/components/activities/check_all_that_apply/CheckAllThatApplyDelivery.tsx

```diff
@@ -65,21 +66,22 @@ const CheckAllThatApply = (props: DeliveryElementProps<CheckAllThatApplyModelSch
   const [hasMoreHints, setHasMoreHints] = useState(props.state.parts[0].hasMoreHints);
   const [selected, setSelected] = useState<ActivityTypes.ChoiceId[]>(
     props.state.parts[0].response === null
-    ? []
-    : props.state.parts[0].response.input.split(' ')
-      .reduce(
-        (acc: ActivityTypes.ChoiceId[], curr: ActivityTypes.ChoiceId) => acc.concat([curr]),
-        []));
+      ? []
+      : props.state.parts[0].response.input.split(' ')
+        .reduce(
+          (acc: ActivityTypes.ChoiceId[], curr: ActivityTypes.ChoiceId) => acc.concat([curr]),
+          []));
 
   const { stem, choices } = model;
 
   const isEvaluated = attemptState.score !== null;
-  const selectedToInput = () => selected.join(' ');
+  const selectionToInput = (newSelection: string | undefined) =>
```

Comment:
Change to allow the "new" selection to be added to the existing selection so that a new selection can be saved before the react state is updated.